homeassistant/components/flunearyou/sensor.py

```diff
@@ -66,32 +111,16 @@ async def async_setup_entry(
     """Set up Flu Near You sensors based on a config entry."""
     coordinators = hass.data[DOMAIN][DATA_COORDINATOR][entry.entry_id]
 
-    sensors: list[CdcSensor | UserSensor] = []
-
-    for (sensor_type, name, icon, unit) in CDC_SENSORS:
-        sensors.append(
-            CdcSensor(
-                coordinators[CATEGORY_CDC_REPORT],
-                entry,
-                sensor_type,
-                name,
-                icon,
-                unit,
-            )
-        )
-
-    for (sensor_type, name, icon, unit) in USER_SENSORS:
-        sensors.append(
-            UserSensor(
-                coordinators[CATEGORY_USER_REPORT],
-                entry,
-                sensor_type,
-                name,
-                icon,
-                unit,
-            )
-        )
-
+    sensors: list[CdcSensor | UserSensor] = [
+        CdcSensor(coordinators[CATEGORY_CDC_REPORT], entry, description)
+        for description in CDC_SENSOR_DESCRIPTIONS
+    ]
+    sensors.extend(
+        [
+            UserSensor(coordinators[CATEGORY_USER_REPORT], entry, description)
+            for description in USER_SENSOR_DESCRIPTIONS
+        ]
+    )
```

Comment:
If you add the "category" (`CATEGORY_CDC_REPORT` | `CATEGORY_USER_REPORT`) to the entity description, you can remove this logic and change it into a simple single loop

Reply:
Don't know if that would be useful here. There are two separate classes which are called: `CdcSensor` and `UserSensor`.

Reply:
Right, agreed.